backend/capellacollab/sessions/hooks/jupyter.py

```diff
@@ -77,3 +88,38 @@ def pre_session_termination_hook(
 
     def _determine_base_url(self, username: str):
         return f"{self._jupyter_public_uri.path}/{username}"
+
+    def _get_project_share_volume_mounts(
+        self,
+        db: orm.Session,
+        token: dict[str, t.Any],
+        tool: tools_models.DatabaseTool,
+    ) -> list[operators_models.Volume]:
+        volumes = []
+
+        for model in toolmodels_crud.get_models_by_tool(db, tool.id):
+            if model.configuration and "workspace" in model.configuration:
+                if not auth_injectables.ProjectRoleVerification(
+                    required_role=projects_users_models.ProjectUserRole.USER,
+                    verify=False,
+                )(model.project.slug, token, db):
+                    # No access to the project
+                    continue
+
+                read_only = not auth_injectables.ProjectRoleVerification(
+                    required_role=projects_users_models.ProjectUserRole.USER,
+                    required_permission=projects_users_models.ProjectUserPermission.WRITE,
+                    verify=False,
+                )(model.project.slug, token, db)
```

Comment:
Same as above. Can you move this to a `has_project_write_access` function?

Reply:
Done.